src/libstore/globals.hh

```diff
@@ -970,6 +970,16 @@ public:
           resolves to a different location from that of the build machine. You
           can enable this setting if you are sure you're not going to do that.
         )"};
+
+    Setting<bool> useXDGBaseDirectories{
+        this, false, "use-xdg-base-directories",
+        R"(
+          If set to `true`, Nix will conform to the [XDG base directory specification]
+          for files in $HOME.
+
```

Comment:
```suggestion
Note that this changes the location of some well-known symlinks that Nix creates, which might cause some breakages
```

(or something of that taste. Just make it explicit that it can be dangerous)